ci/run_cudf_polars_polars_tests.sh

```diff
@@ -48,7 +48,9 @@ python -m pytest \
        --cache-clear \
        -m "" \
        -p cudf_polars.testing.plugin \
-       -v \
+       -n 8 \
```

Comment:
Is there a particular reason you chose 8 workers here?  If there aren't concerns about OOMing, this is probably better set to `"${PARALLEL_LEVEL}"` so it's portable if the underlying instance type running this job is changed.
```suggestion
       -n "${PARALLEL_LEVEL}" \
```

Reply:
I think using `PARALLEL_LEVEL` would be too much for this job and other pytest jobs. The GPU that runs these tests is an L4, we might want to bump this a little bit more but not a very high value that `PARALLEL_LEVEL` contains. Previously when I tried to do this GPU memory went into OOM.

Reply:
> 8 workers is because that's what we use for other pytests across all jobs in cudf CI.

Not blocking, just curious if this is the result of experience or if someone chose 8 at some point and it's been repeatedly copied.

Reply:
I played around with this number quite a bit and settled on 8 workers for the old ci infra. Since we now have GPUs with higher memory, I think we can go higher. I have opened a pr here to test it out separately: https://github.com/rapidsai/cudf/pull/18047

Reply:
Usually we use `PARALLEL_LEVEL` at build time rather than runtime since the variable is reflective of the number of CPU cores (`nproc`) not the available GPU memory.